`ecs::define_label!` usage documentation is incomplete

https://docs.rs/bevy/0.18.0/bevy/ecs/macro.define_label.html

The documentation explains how to use the macro to define a trait, but does not explain, directly or via link to other documentation, how to implement the trait. If I flesh out the example with all things that are obvious, I get:

```rust
bevy::ecs::define_label!(MyNewLabelTrait, MY_NEW_LABEL_TRAIT_INTERNER);

#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
struct MyLabel;

impl MyNewLabelTrait for MyLabel {}

fn main() {
    let interned1 = MY_NEW_LABEL_TRAIT_INTERNER.intern(&MyLabel);
    let interned2 = MY_NEW_LABEL_TRAIT_INTERNER.intern(&MyLabel);
    assert_eq!(interned1, interned2);
}
```

```
error[E0046]: not all trait items implemented, missing: `dyn_clone`
 --> src/bin/scratch.rs:6:1
  |
1 | bevy::ecs::define_label!(MyNewLabelTrait, MY_NEW_LABEL_TRAIT_INTERNER);
  | ---------------------------------------------------------------------- `dyn_clone` from trait
...
6 | impl MyNewLabelTrait for MyLabel {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `dyn_clone` in implementation
```

With the help of rust-analyzer and guessing based on the name and signature, I can fill out the trait implementation and get a program that works as I expect:

```rust
impl MyNewLabelTrait for MyLabel {
    fn dyn_clone(&self) -> Box<dyn MyNewLabelTrait> {
        Box::new(*self)
    }
}
```

However, achieving that requires *both* rust-analyzer or a macro expansion tool (to extract the method signature from the macro-generated trait), and experience with Rust types and idioms (to know what to write for `dyn_clone()`). There should be documentation explaining what to do. Perhaps there could also be a function-like macro to implement the boilerplate part of the `impl`:

```rust
impl MyNewLabelTrait for MyLabel {
    // expands to fn dyn_clone...
    impl_label_trait!(MyNewLabelTrait);

    // extra methods defined by args to `define_label!` go here
}
```

Alternatively, perhaps this macro was not intended to be used outside of the Bevy project. In that case, it should be `doc(hidden)`.

I might be interested in writing the expanded documentation and helper macro, but I need to know if my assumptions about how to use `define_label!` are correct.

--- a/crates/bevy_ecs/src/label.rs
+++ b/crates/bevy_ecs/src/label.rs
@@ -54,18 +54,46 @@ where
     }
 }
 
-/// Macro to define a new label trait
+/// Macro to define a new label trait.
 ///
-/// # Example
+/// Each label trait has an associated [`Interner<dyn YourLabelTraitHere>`][crate::intern::Interner]
+/// The trait has an `intern(&self)` method which uses that interner to
+/// produce [`Interned<dyn YourLabelTraitHere>`][crate::intern::Interned] values,
+/// and a `dyn_clone(&self)` method which must be implemented for the system to work.
+///
+/// # Examples
+///
+/// Minimal working example:
 ///
 /// ```
 /// # use bevy_ecs::define_label;
+/// // Defines `trait MyNewLabelTrait` and `static MY_NEW_LABEL_TRAIT_INTERNER`.
+/// // You don’t need to use the interner for anything; just give it a unique name.
 /// define_label!(
 ///     /// Documentation of label trait
 ///     MyNewLabelTrait,
 ///     MY_NEW_LABEL_TRAIT_INTERNER
 /// );
 ///
+/// /// A new label type implementing the new label trait.
+/// #[derive(Clone, Debug, Eq, Hash, PartialEq)]
+/// pub struct MyLabel;
+///
+/// impl MyNewLabelTrait for MyLabel {
+///     // Implementations of the trait must implement the `dyn_clone()` method in this way
+///     // to enable cloning the trait object because `Clone` is not `dyn` compatible.
+///     fn dyn_clone(&self) -> Box<dyn MyNewLabelTrait> {
+///         Box::new(self.clone())
+///     }
+/// }
+///
+/// assert_eq!(MyLabel.intern(), MyLabel.intern());
+/// ```
+///
+/// A label trait defined by this macro can also be given additional methods:
+///
+/// ```
+/// # use bevy_ecs::define_label;
 /// define_label!(
 ///     /// Documentation of another label trait
 ///     MyNewExtendedLabelTrait,
@@ -75,13 +103,63 @@ where
 ///         fn additional_method(&self) -> i32;
 ///     },
 ///     extra_methods_impl: {
-///         // Implementation of the extra methods for Interned<dyn MyNewExtendedLabelTrait>
+///         // Implementation of the extra methods for Interned<dyn MyNewExtendedLabelTrait>,
+///         // which should usually forward to the contained value.
 ///         fn additional_method(&self) -> i32 {
-///             0
+///             (**self).additional_method()
 ///         }
 ///     }
 /// );
+///
+/// #[derive(Clone, Debug, Eq, Hash, PartialEq)]
+/// pub struct MyLabel;
+///
+/// impl MyNewExtendedLabelTrait for MyLabel {
+///     fn dyn_clone(&self) -> Box<dyn MyNewExtendedLabelTrait> {
+///         Box::new(self.clone())
+///     }
+///
+///     fn additional_method(&self) -> i32 {
+///         42
+///     }
+/// }
+///
+/// let interned_label = MyLabel.intern();
+/// assert_eq!(interned_label.additional_method(), 42);
 /// ```
+///
+/// In order to minimize boilerplate for each new label type, you may wish to define a macro to
+/// generate labels. In Bevy’s own traits, this is done by derive macros (e.g.
+/// `derive(ScheduleLabel)`), but it is often sufficient to write a simple, less general
+/// `macro_rules!` macro:
+///
+/// ```
+/// # use bevy_ecs::define_label;
+/// define_label!(
+///     Team,
+///     TEAM_INTERNER
+/// );
+///
+/// macro_rules! define_team {
+///     ($name:ident) => {
+///         #[derive(Clone, Debug, Eq, Hash, PartialEq)]
+///         pub struct $name;
+///
+///         impl Team for $name {
+///             fn dyn_clone(&self) -> Box<dyn Team> {
+///                 Box::new(self.clone())
+///             }
+///         }
+///     }
+/// }
+///
+/// define_team!(Home);
+/// define_team!(Away);
+///
+/// assert_eq!(Home.intern(), Home.intern());
+/// assert_ne!(Home.intern(), Away.intern());
+/// ```
+///
 #[macro_export]
 macro_rules! define_label {
     (

